guppylang/cfg/builder.py

```diff
@@ -492,6 +479,31 @@ def is_functional_annotation(stmt: ast.stmt) -> bool:
     return False
 
 
+def is_py_expression(node: ast.AST) -> PyExpr | None:
+    """Checks if the given node is a compile-time `py(...)` expression and turns it into
+    a `PyExpr` AST node.
+
+    Otherwise, returns `None`.
+    """
+    if (
+        isinstance(node, ast.Call)
+        and isinstance(node.func, ast.Name)
+        and node.func.id == "py"
```

Comment:
Yeah, this is broken. See #520

In see two options:
* Turn `py` into an actual function and use a `CustomCallChecker`
* Disallow naming variables `py` since it's a "keyword"